src/components/m-dropdown/js/drop-down.js

```diff
@@ -148,6 +149,17 @@ class DropDown extends UiEvents {
       lastElementChild.style.height = '';
       remove(node, this.options.isOpenClass);
     }
+    const select = this.wcNode.querySelector(`.${this.options.nativeSelectClass}`);
+    if (select) {
+      const selectedOption = this.wcNode.querySelector(this.options.selectedItemSelector);
+      if (selectedOption) {
+        if (selectedOption.index !== select.selectedIndex) {
+          select.selectedIndex = selectedOption.index;
+        }
+      } else {
+        select.selectedIndex = 0;
```

Comment:
`0` to unset a `<select>` is incorrect.

Please see the docs here:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedIndex

Reply:
thank you Andy, I'll update the PR and change the implementation, I see your point and agree.